### ISSUE TITLE
Translations (looking for contributors)

### ISSUE THREAD

**Issue by nukeop** [nukeop/nuclear#345]
We're looking for contributors who are willing to translate Nuclear to their native languages (or any languages that they know well). Currently @charjac has done all the work required to add new translations, and also translated all the strings to French.

New locales should be placed here: https://github.com/nukeop/nuclear/tree/master/app/locales

**Languages currently available:**
- English
- French (by @charjac)
- Dutch (by @Vistaus)
- Danish (by @Hansen1992)
- Spanish (by @mlucas94 )
- Polish (by @kazimierczak-robert and @gradzka)

**Proposed:**
- Chinese (simplified) (by @lizzyd710)
- Russian (by @ramstore07)

**Comment by charjac** [comment]
When adding translations, a few things need to be done:
 - add the json file with the correct name in app/locales and keep it empty
 - update the i18n script in package.json by adding your locale 
```
{
  ...
  "scripts": {
    ...
     "i18n": "sync-i18n --files 'app/locales/*.json' --primary en --languages fr [your locale] --space 2"
  } 
  ...
}
```
 - run the command
```
npm run i18n
```

you can now translate your locale file from english to your language.

**Comment by Hansen1992** [comment]
Hey @nukeop, I would like to help you translate Nuclear from English to Danish if this has any interest?

**Comment by nukeop** [comment]
Absolutely, in general if you don't see a language on the list above, we want it added. Thanks for the help.

**Comment by lizzyd710** [comment]
What kind of things or phrases would you like to be translated? I'm a native English speaker but I took a few years of Chinese (Mandarin, so I write simplified). I may not be able to do much, but I would be happy to start and do as much as I can.

**Comment by nukeop** [comment]
You can see a list of all strings that have to be translated here: https://github.com/nukeop/nuclear/blob/master/app/locales/en.json

Even if you only translate a part of it, any missing strings fall back on default English. And some is better than nothing.

**Comment by lizzyd710** [comment]
> 
> 
> When adding translations, a few things need to be done:
> 
>     * add the json file with the correct name in app/locales and keep it empty
> 
>     * update the i18n script in package.json by adding your locale
> 
> 
> ```
> {
>   ...
>   "scripts": {
>     ...
>      "i18n": "sync-i18n --files 'app/locales/*.json' --primary en --languages fr [your locale] --space 2"
>   } 
>   ...
> }
> ```
> 
>     * run the command
> 
> 
> ```
> npm run i18n
> ```
> 
>     * edit the file app/constants/settings.js to add your local in user's choice
> 
> 
> ```js
>   {
>     name: 'language',
>     category: 'program-settings',
>     type: settingType.LIST,
>     prettyName: 'language',
>     placeholder: 'language-placeholder',
>     options: ['en', 'fr', <your locale>],
>     default: undefined
>   }
> ```
> 
>     * edit the app/i18n.js file
> 
> 
> ```js
> ...
> import en from './locales/en.json';
> import <your locale> from './locales/<your locale>.json';
> ...
>     resources: {
>       en,
>       ...
>       <your locale>
>     },
> ...
> ```
> 
> you can now translate your locale file from english to your language.

(This might not be the appropriate place for this, so I apologize and let me know if/where you want me to move this.) Do I need to run `npm run i18n` before editing app/constants/settings.js and app/i18n.js? I'm having some issues with setting up the NPM dev environment for working on this repo, and I'm wondering if I could just edit the files in the meantime to try and get started on the translations while I troubleshoot.

**Comment by charjac** [comment]
npm run i18n just fill the new json file with english translation. 
you can copy paste en.json content to <your locale>.json, it will do the same.

wich error do you get when you setup dev environment ?

**Comment by lizzyd710** [comment]
(If I should have posted this in pastebin or something else, let me know. I know different repos have different protocols for this type of thing.)
![image](https://user-images.githubusercontent.com/12602768/59435268-39d26000-8dbb-11e9-8f46-16ea6fc3b5b4.png)

**Comment by charjac** [comment]
arf windows :'( :'(

There is a few things you can try:
 - run your cmd as administrator
 - delete node_modules and then run npm install
 - maybe try gitbash ?
 - help !!! :)

**Comment by nukeop** [comment]
I think it's a matter of running `npm install` first.

**Comment by lizzyd710** [comment]
Yeah, I reinstalled and it loaded, it's just stuck on the nuclear logo screen. Would I be able to run the i18n command though, or do I need to figure out why the app isn't loading first? (Again, sorry for clogging up this issue. Is there a better place for me to go for this, or is here fine?)

**Comment by lizzyd710** [comment]
So, I was afraid this would be an issue with translating to Chinese, but even when I save the `zh.json` file with Unicode encoding it doesn't properly parse...how should I go about trying to handle this?

**Comment by nukeop** [comment]
Talking about it here is fine. Could you start a pull request with the changes you've made so far so I can pull them and check? Maybe it'll work fine for me, and even if it doesn't I'll be in a better position to judge what's wrong.

**Comment by ramstore07** [comment]
Hey there! I would like to translate it to Russian! Let me know what you think!

**Comment by nukeop** [comment]
Sure, this is a great idea. Let me know if there are any problems.

**Comment by lizzyd710** [comment]
> 
> 
> Talking about it here is fine. Could you start a pull request with the changes you've made so far so I can pull them and check? Maybe it'll work fine for me, and even if it doesn't I'll be in a better position to judge what's wrong.

I think I figured it out. I saved it with Unicode encoding but I should have saved it as UTF-8. *sigh*

Now I just have to figure out how to actually get the program to get past the logo and I can test and submit a pull request.

**Comment by nukeop** [comment]
This is coming along really, really well. We've got 9 languages already, and it's barely been two weeks.

**Comment by bh-c** [comment]
Hi there! Though it's my first time to contribute to a GitHub project, I would like to translate it into Chinese(Traditional). Can I give it a try?

**Comment by nukeop** [comment]
Absolutely, everyone is welcome to contribute.

**Comment by avirec** [comment]
Hi :)
I'll start to translate it to hebrew

**Comment by nukeop** [comment]
Thank you. Let me know if it causes any problems, I don't know if RTL languages will be rendered correctly.

**Comment by avirec** [comment]
Just to make sure (and sorry for the newbie question) - should I open a branch and work on the relevant files locally, and when I finish - to do a pull request?

**Comment by nukeop** [comment]
First you need to fork the repository using the button in the top right corner. This creates your own copy. Inside that copy, you can either create a new branch, or work on the master branch (up to your preference). After you're finished, you'll have the option to create a new pull request from the branch you worked on to my repo.

**Comment by avirec** [comment]
> Thank you. Let me know if it causes any problems, I don't know if RTL languages will be rendered correctly.

Is there a way to check the RTL before I'll complete the translation?

**Comment by nukeop** [comment]
You can try starting the development version and selecting your new language in the settings.
```shell
$ npm install
$ npm run watch && npm run electron:dev
```

**Comment by avirec** [comment]
Thanks :)
Well, I've translated most of the strings but some of the strings contain words or phrases that should remain be in english, even after the translation (like "API", "Last.fm" and so on). In these cases, when the string contain english and hebrew both, There is a RTL issue.
So I think that these strings will remain in english and in the future we can add some css fixes for RTL version (and then I'll translate the mixed strings).
What do you think? :)

**Comment by nukeop** [comment]
Ok, I'll look into it and how to handle it best.

**Comment by nukeop** [comment]
For reference: https://langsolinc.com/rtl-language-direction-in-css-basics/

**Comment by hedyhyry** [comment]
Hi I just translated some phrases into Chinese, and made a pull request.

**Comment by nukeop** [comment]
Thank you, appreciated.

**Comment by nukeop** [comment]
@avirec 
https://www.i18next.com/overview/api#dir

Looks like it can be automatically detected whether or not a given language is RTL. However, this will need additional styles applied, not just to text but also to e.g. reverse the order in which buttons and labels appear in some contexts. It is likely that this will have to be developed as a separate feature.

**Comment by avirec** [comment]
@nukeop 
Great. I can do it (I mean - the css chenges for the RTL version)

**Comment by weninglaks** [comment]
Hi!  I'm new to github but I'd like to help to translate it to Indonesian. How do I start?

**Comment by nukeop** [comment]
@weninglaks have a look here: https://github.com/nukeop/nuclear/wiki/Translating

**Comment by Zhenye-Na** [comment]
Has the translation of the Chinese version already finished? If not, I would love to contribute to it.

Thanks!

**Comment by nukeop** [comment]
@Zhenye-Na Nope, not yet. You can see all the needed strings here: https://github.com/nukeop/nuclear/blob/master/app/locales/zh.json

**Comment by Zhenye-Na** [comment]
@nukeop Great! May I work on to finish it right now? Also, are there any guidelines to start with?

**Comment by nukeop** [comment]
Sure, feel free to translate whatever's left. Normally you'd have to add some code to enable a new translation but this one's already in the program so you don't need to do anything other than the translation itself.

**Comment by zhr1996** [comment]
Hi I could help with translating Chinese! Is there anything that I can contribute to or help with?

**Comment by nukeop** [comment]
Thanks for the offer @zhr1996 ! You can find more info here: https://github.com/nukeop/nuclear/wiki/Translating

In the case of Chinese, you only need to translate the remaining strings, since all the other parts are already completed.

**Comment by zhr1996** [comment]
@nukeop Thanks! I'll try to translate them and pull request!

**Comment by giftofgrub** [comment]
Hi, I can contribute to translating it in Filipino (Tagalog).  Is the project still accepting translations?

**Comment by nukeop** [comment]
Yep, of course! We don't have Filipino yet. It would be great to add it.

**Comment by MartinTuroci** [comment]
Hi, gonna work on translating into Slovak language.

**Comment by dexterastin** [comment]
Hey there! I would like to translate it to Korean! Let me know what you think!

**Comment by nukeop** [comment]
Sure, there is no Korean translation yet, it would be a fantastic addition.

**Comment by jqiallnatural** [comment]
Hello, I can try to finish the remaining translation for Chinese, since there is already a zh.json file, I can just make the changes in zh.json then upload the pull request correct? do i still need to update the i18n script in packages/app/package.json by adding your locale??? thank you kindly

**Comment by nukeop** [comment]
You only need to translate the remaining strings. There will be more gaps as we add new features, so there's often something new to add.

**Comment by nukeop** [comment]
Sure, we don't have Greek yet.

**Comment by pozda** [comment]
I'll add Croatian translation! Just forked the repo!

**Comment by nukeop** [comment]
Thank you, this is great!

**Comment by cjola002-xamk** [comment]
Hello! I would like to start working on the Finnish translation if nobody haven't started it yet?

**Comment by nukeop** [comment]
Finnish isn't supported yet so you can do it, thanks for the help.

**Comment by cjola002-xamk** [comment]
> Finnish isn't supported yet so you can do it, thanks for the help

Nice, thanks! I'll start working on it today.

**Comment by azumukupoe** [comment]
Hi, I can help with Japanese

**Comment by nukeop** [comment]
Thanks, that would be great. Please follow the translation guide.

**Comment by teamzamong** [comment]
#888 hi, i updated the korean translation & added a korean translation for readme.

**Comment by nukeop** [comment]
From now on, new translations should be added here: https://crowdin.com/project/nuclear

**Comment by UnixCro** [comment]
I have a very good command of the German language. On request, I can check German sentences for grammar, spelling, style & syntax.

**Comment by nukeop** [comment]
@UnixCro You can use this link to join as a proofreader: https://crowdin.com/project/nuclear/invite?d=052515j696g5i5b3s4k493t4j4

**Comment by amendezm** [comment]
@nukeop I would like to work in an spanish readme translation. What are the steps to follow?

**Comment by nukeop** [comment]
Spanish is at 100% right now, so no need to do anything: https://crowdin.com/project/nuclear

**Comment by amendezm** [comment]
@nukeop I mean the readme file of the repo

![image](https://user-images.githubusercontent.com/26444448/128079404-5641c104-7971-4859-acee-b85195187da5.png)
![image](https://user-images.githubusercontent.com/26444448/128079478-4515ec85-3c13-4b0a-a15f-fa0c397f27f8.png)

I don't see spanish there

**Comment by nukeop** [comment]
In this case, just create a file named README-[locale code].md, and add a link to it in the English readme.

**Comment by amendezm** [comment]
@nukeop I just open a PR for this

**Comment by SecularSteve** [comment]
I just translated Croatian, Dutch and German on CrowdIn. With a few additional checks, especially in Croatian.

Is it also possible to add Serbian and Bosnian?

**Comment by nukeop** [comment]
Awesome, I'll add them. How do you know so many languages?

**Comment by SecularSteve** [comment]
> 

Bosnian, Croatian and Serbian are all much of the same, but with slight dialect differences. The only reason they're proclaimed different today is because of complicated socio-political reasons.

I already had experience with English and German, so learning Dutch was automatically a breeze.

**Comment by Altela** [comment]
Hi @nukeop 
Just translated and finished Indonesian language.
Let me know if there's anything to help! i'd be glad to do so :D

**Comment by FahimFBA** [comment]
I was hoping for adding a translation in my native language, Bengali.

**Comment by nukeop** [comment]
Ok, I added Bengali on Crowdin, you should be able to start translating.

**Comment by rm-NoobInCoding** [comment]
Hi
Please add Persian language to Crowdin.
Thanks

**Comment by nukeop** [comment]
Thanks for the suggestion, I added it.

**Comment by Aftzalanc** [comment]
Hi @nukeop! Could I help with the Greek translation? It's my first time contributing to an open source project, so I might have a few questions, but I really want to help in any way I can 🙂 .

**Comment by nukeop** [comment]
Hi, sure, although Greek is already 91% complete.

**Comment by MagicLike** [comment]
Does German need any proofreading or is it completely translated?

**Comment by UnixCro** [comment]
As a native German speaker, I corrected a few things back then. Many third-party translations were utter nonsense. If I were you, I would look and see what else you can find, since the owner cannot test German and therefore cannot give you the answer you expect...

**Comment by Aftzalanc** [comment]
Thank you @nukeop, I'll start and hopefully now it will reach 100%. I could also proofread the existing translations or try to translate the README.md (since I didn't see a Greek version) if that would be helpful.

**Comment by nukeop** [comment]
It always helps when native speakers confirm translations. There's around 25 languages or even more, and I have no choice but to accept whatever is contributed since its hard to estimate correctness.

**Comment by Aftzalanc** [comment]
Hello! So, just to make sure I have fully understood how crowdin works, if I have a translation for a string or I think one of the suggestions is correct, I just use and save it, right? Also, while proofreading, should I 'approve' every accurate translation?

**Comment by nukeop** [comment]
The first part is correct, if you want to proofread, feel free to approve as many strings you want, the more are approved by native speakers, the more confident we can be in the translation.

**Comment by Aftzalanc** [comment]
Hi. I seem to have stumbled across a somewhat minor dilemma. There are some words/phrases that could be translated in Greek, but in everyday life their translations are not used (they sound kind of formal/odd) and the original English words are preferred instead. For example, Plugins, Playback etc. I can use the translations if you believe it to be best though. 
I appreciate your patience with all these questions and your guidance 🙂

**Comment by nukeop** [comment]
I think in this case it's better to use the original English, pretty sure that's what's used in other languages as well.

**Comment by NohaFahmi** [comment]
@nukeop Hi, can I create a Readme in Arabic language and add it to the English Readme?

**Comment by nukeop** [comment]
Sure you can!

**Comment by notashwinii** [comment]
Hi @nukeop, I was hoping to add translations to my native language, Nepali!

**Comment by nukeop** [comment]
Ok, I added Nepali. You can start now.

**Comment by nukeop** [comment]
This issue is from an older version of Nuclear that has been replaced with a complete rewrite. None of the old issues apply to the new codebase.

The old code is preserved on the [legacy/electron](https://github.com/nukeop/nuclear/tree/legacy/electron) branch. If you have feedback about the new version, head to [Discussions](https://github.com/nukeop/nuclear/discussions).
